packages/middleware-cors/src/configureHeaders.ts

```diff
@@ -0,0 +1,116 @@
+import { HttpRequest, HttpResponse, HttpMethod, HttpStatus } from '@marblejs/core';
+
+import { checkOrigin } from './checkOrigin';
+import { CORSOptions } from './middleware';
+import { capitalize, isString } from './util';
+
+interface ConfiguredHeader {
+  key: string;
+  value: string;
+}
+
+enum AccessControlHeader {
+  Origin = 'Access-Control-Allow-Origin',
+  Methods = 'Access-Control-Allow-Methods',
+  Headers = 'Access-Control-Allow-Headers',
+  Credentials = 'Access-Control-Allow-Credentials',
+  MaxAge = 'Access-Control-Max-Age',
+  ExposeHeaders = 'Access-Control-Expose-Headers',
+}
+
+export const applyHeaders = (
+  headers: ConfiguredHeader[],
+  res: HttpResponse,
+): void => {
+  headers.forEach(({ key, value }) => {
+    res.setHeader(key, value);
+  });
+};
+
+export function configurePreflightResponse(
```

Comment:
`configurePreflightResponse` and `configureResponse` I would extract into separate files. It would be more readable and the body of the functions we could also extract into smaller functions.